meshtastic/powermon/__init__.py

```
@@ -1,11 +1,58 @@
+    dependency_name: str,
+    import_error: ImportError,
+) -> type[PowerSupply]:
+    """Return a placeholder backend class that raises a clear dependency error."""
+
+    class _MissingOptionalBackend(PowerSupply):
+        def __init__(self, *args: object, **kwargs: object) -> None:
+            _ = (args, kwargs)
+            raise ImportError(
+                f"{backend_name} requires optional dependency {dependency_name!r}. "
+                "Install Meshtastic with powermon extras to use this backend."
+            ) from import_error
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)